src/utils/label.ts

```diff
@@ -87,20 +90,44 @@ function hasAncestorCustomElements(label: HTMLCalciteLabelElement, componentEl:
 export function connectLabel(component: LabelableComponent): void {
   const labelEl = findLabelForComponent(component.el);
 
-  if (!labelEl || onLabelClickMap.has(labelEl)) {
+  const addClickEventListenerToComponentLabel = () => {
+    component.labelEl = labelEl;
+    const boundOnLabelClick = onLabelClick.bind(component);
+    onLabelClickMap.set(component.labelEl, boundOnLabelClick);
+    component.labelEl.addEventListener(labelClickEvent, boundOnLabelClick);
+  };
+
+  const boundOnLabelConnected = onLabelConnected.bind(component);
+  const boundOnLabelDisconnected = onLabelDisconnected.bind(component);
+
+  if (onLabelClickMap.has(labelEl)) {
     return;
   }
 
-  component.labelEl = labelEl;
-  const boundOnLabelClick = onLabelClick.bind(component);
-  onLabelClickMap.set(component.labelEl, boundOnLabelClick);
-  component.labelEl.addEventListener(labelClickEvent, boundOnLabelClick);
+  if (labelEl) {
+    addClickEventListenerToComponentLabel();
+    unlabeledComponents.delete(component);
+    document.removeEventListener(labelConnectedEvent, boundOnLabelConnected);
+    document.body.addEventListener(labelDisconnectedEvent, boundOnLabelDisconnected);
+  }
+
+  if (!labelEl) {
+    unlabeledComponents.add(component);
+    document.addEventListener(labelConnectedEvent, boundOnLabelConnected);
+    document.body.removeEventListener(labelDisconnectedEvent, boundOnLabelDisconnected);
+  }
 }
 
 /**
- * Helper to tear down label interactions on disconnectedCallback.
+ * Helper to tear down label interactions on disconnectedCallback on labelable components.
  */
 export function disconnectLabel(component: LabelableComponent): void {
+  const boundOnLabelConnected = onLabelConnected.bind(component);
+  const boundOnLabelDisconnected = onLabelDisconnected.bind(component);
+  unlabeledComponents.delete(component);
+  document.removeEventListener(labelConnectedEvent, boundOnLabelConnected);
+  document.body.removeEventListener(labelDisconnectedEvent, boundOnLabelDisconnected);
```

Comment:
`document.body?` just incase.

It looks like the above event just uses `document` and not `document.body`. Maybe we should be consistent here?

Reply:
I'm not sure if we should emit on the document or the body. Maybe the document is ok?

Reply:
I think it should be document, unless the listener should be added to a more specific element.

Reply:
document seems best 👍